### PR TITLE
feat(ConfigProvider): attach DisableContext to ConfigProvider

### PR DESCRIPTION
**Pull request by xliez**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/39545

### 💡 Background and solution

目前自定义表单组件中要获取父级 Form 组件的禁用状态的话，需要从dist代码中获取 DisabledContext ，希望能将这个 API 导出以方便使用。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      attach DisableContext to ConfigProvider for accessing the API easily     |
| 🇨🇳 Chinese |     将 DisableContext 挂到 ConfigProvider 以方便调用     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by xliez** [comment]
I found argos checks failed, there are  76 warnings not related with my commits

**Comment by li-jia-nan** [comment]
> I found argos checks failed, there are 76 warnings not related with my commit.
> 
> And why my pr branch is compared with master branch? I think it should compare with feature branch.
> 
> welcome your commets @MadCcc @li-jia-nan , thanks

Changes approved~

**Comment by xliez** [comment]
这个PR可以merge么？

**Review by yoyo837 — APPROVED** [review]

File: components/form/demo/disabled.tsx [hunk 1]
```diff
@@ -13,11 +13,22 @@ import {
...
 
+const CustomizedFormControls = () => {
+  const disabled = useContext(DisabledContext);
+  return (
+    <Form.Item label="custom field">
+      <Input disabled={disabled} />
+    </Form.Item>
+  );
+};
+
```
Comment (MadCcc) [inline components/form/demo/disabled.tsx:31]: 这个自定义组件还是用到了 antd 的 Input，本身就会受 context 影响，可以重新写一下

File: components/config-provider/index.tsx [hunk 2]
```diff
@@ -325,5 +327,6 @@ const ConfigProvider: React.FC<ConfigProviderProps> & {
 ConfigProvider.ConfigContext = ConfigContext;
 ConfigProvider.SizeContext = SizeContext;
 ConfigProvider.config = setGlobalConfig;
+ConfigProvider.DisabledContext = DisabledContext;
```
Comment (zombieJ) [inline components/config-provider/index.tsx:330]: 这样不行，export 零零散散的 context 太多了。我写个 RFC 直接提供一个统一的 Hooks。

@xliez 等我一下。

Comment (zombieJ) [reply components/config-provider/index.tsx:330]: #40025

**Pull request closed without merging**